src/main/java/step3/car/RacingCar.java

```diff
@@ -1,21 +1,52 @@
 package step3.car;
 
-import step3.game.RacingGame;
+import step3.strategy.MoveStrategy;
+import step3.strategy.PrintMarkStrategy;
+
+import java.util.Random;
 
 public class RacingCar implements Car {
-    private final StringBuilder skidMark = new StringBuilder();
+    private int progress;
+    private MoveStrategy moveStrategy;
+    private final StringBuilder sb = new StringBuilder();
+    private static final Random random = new Random();
 
     public RacingCar() {
+        moveStrategy = MoveStrategy.defaultStrategy(random);
+        progress = 0;
+    }
+
+    public RacingCar(MoveStrategy moveStrategy) {
+        progress = 0;
+        this.moveStrategy = moveStrategy;
     }
 
+
     @Override
-    public void go(RacingGame game) {
-        if (allowMove(game.getNumber())) {
-            skidMark.append("-");
+    public void go() {
+        if (allowMove()) {
+            progress += 1;
```

Comment:
```suggestion
            move();
```
메소드로 행위를 나타내는것과 코드로 풀어쓰는것은 큰 차이가 있습니다 😄 
중요한 행위는 다른 사람이 보더라도 알아보기 쉽게 구현하는것이 중요합니다.